src/agenteval/cli.py

```diff
@@ -1045,5 +1040,215 @@ def eval_command(
 
 cli.add_command(eval_command)
 
+
+@dataclass
+class WithinRepoPathComponents:
+    hf_config: str
+    split: str
+    submission_name: str
+
+    def to_within_repo_result_path(self):
+        return f"{self.hf_config}/{self.split}/{self.submission_name}.json"
+
+    def to_within_repo_submission_dir(self):
+        return f"{self.hf_config}/{self.split}/{self.submission_name}"
+
+    def to_within_repo_submission_summary_dir(self):
+        return f"summaries/{self.hf_config}/{self.split}/{self.submission_name}"
+
+    def to_within_repo_submission_eval_config(self):
+        return f"{self.to_within_repo_submission_dir()}/{EVAL_CONFIG_FILENAME}"
+
+    def to_within_repo_submission_submission_metadata(self):
+        return f"{self.to_within_repo_submission_dir()}/{SUBMISSION_METADATA_FILENAME}"
+
+    def to_within_repo_submission_scores(self):
+        return f"{self.to_within_repo_submission_summary_dir()}/{SCORES_FILENAME}"
+
+    def within_repo_submission_patterns(self):
+        return [
+            self.to_within_repo_submission_eval_config(),
+            self.to_within_repo_submission_submission_metadata(),
+            self.to_within_repo_submission_scores(),
+            f"{self.to_within_repo_submission_dir()}/*.eval",
+        ]
+
+    @staticmethod
+    def from_within_repo_result_path(result_path):
+        [hf_config, split, filename] = result_path.split("/")
+        suffix = ".json"
+        assert filename.endswith(suffix)
+        submission_name = filename[: -len(suffix)]
+        return WithinRepoPathComponents(
+            hf_config=hf_config, split=split, submission_name=submission_name
+        )
+
+    @staticmethod
+    def from_within_repo_submission_path(submission_path):
+        [hf_config, split, submission_name] = submission_path.split("/")
+        return WithinRepoPathComponents(
+            hf_config=hf_config, split=split, submission_name=submission_name
+        )
+
+
+@click.command(name="copy", help="Copy a result from one results HF repo to another.")
+@click.argument("result_urls", nargs=-1, required=True, type=str)
+@click.option(
+    "--target-submissions-repo",
+    default=None,
+    required=False,
+    help="Provide this if you also want to copy the underlying submission to another submissions HF repo.",
+)
+@click.option(
+    "--target-results-repo",
+    default=None,
+    required=True,
+)
+@click.option(
+    "--read-public-logs-field",
+    is_flag=True,
+    default=False,
+    help="Provide this if the source results have log urls in logs_url_public.",
+)
+@click.option(
+    "--write-public-logs-field",
+    is_flag=True,
+    default=False,
+    help="Provide this if the target results should have log urls in logs_url_public.",
+)
+def copy_command(
```

Comment:
Example command, if you wanted to copy something over from the internal repos to the public ones.
```
agenteval copy --target-submissions-repo "allenai/asta-bench-submissions" --target-results-repo "allenai/asta-bench-results" --write-public-logs-field  "hf://allenai/asta-bench-internal-results/1.0.0/test/aakanksha19_Asta_Table_Synthesis__GPT-4.1__2025-07-11T21-23-03.json"  "hf://allenai/asta-bench-internal-results/1.0.0/test/aakanksha19_Asta_Table_Synthesis__Pro-2.5__2025-07-12T06-06-53.json"
```